src/game/Board.java

```diff
@@ -0,0 +1,376 @@
+		intersecs[i].setMark(Mark.U);
+		for (Intersec neighbour : intersecs[i].getNeighbours()) {
+			neighbour.addLiberty(intersecs[i]);
+		}
+	}
+
+	/**
+	 * Removes a chain from the board.
+	 */
+	public void removeChain(Chain chain) {
+		for (Intersec stone : chain.getStones()) {
+			removeStone(stone.getCol(), stone.getRow());
+		}
+	}
+
+	/**
+	 * Put a stone on the board at the provided column and row. Create a chain with
+	 * this stone. Add to this chain any neighbouring chains of the same colour.
+	 * Remove any stones of the opponent player that have no liberties anymore.
+	 * Remove any stones of the player who put the stone that have no liberties
+	 * anymore. Finally updates the board situations with this new board situation
+	 * (for enforcing the Ko-rule).
+	 * 
+	 * @throws ExitProgram if some player (client) chooses an intersection for his
+	 *                     next stone that already has a stone on there
+	 */
+	public void putStone(int col, int row, Mark mark) throws ExitProgram {
+		this.firstStone = false;
+		int i = coorToInt(col, row);
+		if (!isUnoccupied(i)) {
+			throw new ExitProgram("Cannot place a stone at an intersection where there already is one!");
+		}
+		addStone(col, row, mark);
+		Chain chain = new Chain(intersecs[i], mark);
+		for (Intersec neighbour : intersecs[i].getNeighbours()) {
+			if (neighbour.getMark() == mark && !neighbour.getChain().equals(chain)) {
+				chain.joinChain(neighbour.getChain());
+			}
+		}
+		for (Intersec neighbour : intersecs[i].getNeighbours()) {
+			if (neighbour.getMark() == mark.other() && neighbour.getChain().chainLib() == 0) {
+				removeChain(neighbour.getChain());
+			}
+		}
+		if (intersecs[i].getChain().chainLib() == 0) {
+			removeChain(intersecs[i].getChain());
+		}
+		addBoardSituation(this.toString());
+	}
+
+	/**
+	 * After the game has ended (either board full or two consecutive passes in a
+	 * row), the score will be counted by calling this method. For all intersections
+	 * it determines whether there is a black or white stone, and if so the
+	 * corresponding score is incremented. It also determines to which colour empty
+	 * areas belong. Finally it updates the scores of both players by adding up all
+	 * empty intersections that belong to only their are.
+	 */
+	public void countScore() {
+		blackScore = 0;
+		whiteScore = 0.5;
+		scoringBlack();
+		for (int i = 0; i < intersecs.length; i++) {
+			intersecs[i].setChecked(false);
+		}
+		scoringWhite();
+		for (int i = 0; i < intersecs.length; i++) {
+			if (intersecs[i].getAreaColor() == Mark.B) {
+				blackScore = blackScore + 1;
+			} else if (intersecs[i].getAreaColor() == Mark.W) {
+				whiteScore = whiteScore + 1;
+			}
+		}
+	}
+
+	/**
+	 * Gives a point to black for each black stone on the board. Calls the
+	 * areaBlack(Intersec intersec) method on each intersection with a black stone
+	 * on there.
+	 */
+	private void scoringBlack() {
+		for (int i = 0; i < intersecs.length; i++) {
```

Comment:
In this class your doing the "loop over all intersections" a lot. (Either in this way, or even with a double column/row loop.) You could probably have extracted to a single method.